packages/react/src/components/AmplifyProvider/index.tsx

```diff
@@ -21,16 +21,23 @@ export function AmplifyProvider({
 }: AmplifyProviderProps) {
   const webTheme = createTheme(theme);
   const { name, cssText } = webTheme;
+  React.useEffect(() => {
+    if (document && document.documentElement) {
+      document.documentElement.setAttribute('data-amplify-theme', name);
+      document.documentElement.setAttribute(
+        'data-amplify-color-mode',
+        colorMode
+      );
+    }
+  }, [theme, colorMode]);
   return (
     <AmplifyContext.Provider
       value={{
         theme: webTheme,
       }}
     >
       <IdProvider>
-        <div data-amplify-theme={name} data-amplify-color-mode={colorMode}>
-          {children}
-        </div>
+        <div>{children}</div>
```

Comment:
Do we need the `<div>` here? Not blocking, just thinking out loud.

Reply:
+1